cpg_infra/abstraction/gcp.py

```diff
@@ -31,34 +31,41 @@ def name():
     def __init__(
         self, config: CPGInfrastructureConfig, dataset_config: CPGDatasetConfig
     ):
-        super().__init__(config, dataset_config)
-        self.region = 'australia-southeast1'
+        super().__init__(config=config, dataset_config=dataset_config)
+        self.region = dataset_config.gcp.region or config.gcp.region
 
     @cached_property
     def organization(self):
         return gcp.organizations.get_organization(domain=self.config.domain)
 
     @cached_property
     def project_id(self):
-        return gcp.organizations.get_project().project_id
+        return self.dataset_config.gcp.project
+
+    @cached_property
+    def project(self):
+        return gcp.organizations.get_project(self.project_id)
 
     def get_dataset_project_id(self):
         return self.project_id
 
-    # region SERVICES
+    def finalise(self):
+        # Make sure this API is initialised somewhere
```

Comment:
Would this pattern also fix https://centrepopgen.slack.com/archives/C03FZL2EF24/p1667871645464889?thread_ts=1667870590.617809&cid=C03FZL2EF24?

Reply:
No, because you can't create the dependency. The better news is you only need to enable it on the project that Pulumi operates on, you actually don't need to on sub-projects AFAIK, because it uses the parent project's API.